src/lib/y2security/lsm/config.rb

```diff
@@ -0,0 +1,152 @@
+      # fallback in case that no one is selected
+      def propose_default
+        log.info("The settings are #{product_feature_settings.inspect}")
+        selected = product_feature_settings.fetch(:default, "apparmor")
+
+        select(selected)
+      end
+
+      # Select the module with the given id when it is selectable.
+      #
+      # @param id [String,Symbol] LSM id to be selected
+      def select(id)
+        @selected = selectable.find { |m| m.id == id.to_sym }
+        log.info @selected ? "Selected LSM with id: #{id}" : "No LSM selected with id: #{id}"
+        @selected
+      end
+
+      # Convenience method to obtain a list of the supported and selectable modules
+      #
+      # @return [Array<Y2Security::LSM::Base] array of supported and selectable LSMs
+      def selectable
+        supported.select(&:selectable?)
+      end
+
+      # Returns the needed patterns for the selected LSM or an empty array if no one is selected
+      #
+      # @return [Array<Sting>]
+      def needed_patterns
+        return [] unless selected
+
+        selected.needed_patterns
+      end
+
+      # Convenience method to save the configuration for the selected LSM
+      def save
+        return false unless selected
+
+        selected.save
+      end
+
+      # Obtains the supported and active Linux Security Major Module from the running system
+      #
+      # @return [Y2Security::LSM::Base]
+      def from_system
+        active.first
+      end
+
+      # Return an array with the supported and active Linux Security Major Modules
+      #
+      # @return [Array<Y2Security::LSM::Base>]
+      def active
+        return [] unless Yast::Stage.normal
+
+        modules = Yast::SCR.Read(Yast.path(".target.string"), RUNNING_PATH)
+        modules.split(",").each_with_object([]) do |name, result|
+          supported_module = supported.find { |m| m.id.to_s == name }
+          result << supported_module if supported_module
+        end
+      end
+
+      # Returns whether the LSM is configurable during installation or not based in the control file
+      # declaration. It returns false in case it is WSL
+      #
+      # @return [Boolean] true if LSM is configurable during the installation; false otherwise
+      def configurable?
+        return @configurable unless @configurable.nil?
+        return false if Yast::Arch.is_wsl
+
+        @configurable = product_feature_settings[:configurable] || false
+      end
+
+      # Returns the values for the LSM setting from the product features
+      #
+      # @return [Hash{Symbol => Object}] e.g., { default: :selinux, selinux: { "selectable" => true }}
+      #   a hash holding the LSM options defined in the control file;
+      #   an empty object if no settings are defined
+      def product_feature_settings
+        return @product_feature_settings unless @product_feature_settings.nil?
+
+        settings = Yast::ProductFeatures.GetFeature("globals", "lsm").dup
```

Comment:
NP: Of course, it does not hurt, but I think `.dup` is not needed, since [`Yast::ProductFeatures.GetFeature`](https://github.com/yast/yast-yast2/blob/3c6c8d316da67146aeab23cc325d1f84062c5d62/library/control/src/modules/ProductFeatures.rb#L208-L218) uses [`Ops.get`](https://github.com/yast/yast-ruby-bindings/blob/c5de99becfd9b154e995f4a615b8748923c3a088/src/ruby/yast/ops.rb#L112) which returns a [`Yast.deep_copy`](https://github.com/yast/yast-ruby-bindings/blob/c5de99becfd9b154e995f4a615b8748923c3a088/src/ruby/yast/yast.rb#L55-L97)

Which leads me to think if worth a class method for getting the lsm settings and use it here and also in Y2Security::LSM::Base#product_feature_settings

What do you think?